.github/workflows/test-and-publish-nightly.yml

```diff
@@ -94,41 +95,28 @@ jobs:
               make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc coverage -j
             ;;
           esac
-      - name: Upload test logs for default architecture
+      - name: Upload test logs
         uses: actions/upload-artifact@v4
 
         if: always() # we always want to upload test logs, especially when tests fail
         with:
-          name: test-logs-${{ matrix.runson.display }}-${{ matrix.java-version }}
+          name: test-logs-${{ matrix.runson.display }}-${{ matrix.java-version }}${{ matrix.architecture && '-' || '' }}${{ matrix.architecture }}
           path: |
             build/test/logs/
             hs_err*.log
       - name: Upload coverage report
         uses: actions/upload-artifact@v4
 
         with:
-          name: test-coverage-${{ matrix.runson.display }}-${{ matrix.java-version }}
+          name: test-coverage-${{ matrix.runson.display }}-${{ matrix.java-version }}${{ matrix.architecture && '-' || '' }}${{ matrix.architecture }}
           path: build/test/coverage/
-      - name: Test macOS x64
-        if: ${{ matrix.runson.name == 'macos-14' }}
-        run: JAVA_HOME=$JAVA_HOME_${{ env.JAVA_VERSION }}_X64 make test
-
       - name: Upload async-profiler binaries to workflow
         uses: actions/upload-artifact@v4
 
-        if: env.RELEASE_FROM_JAVA_VERSION == matrix.java-version
+        if: ${{ env.RELEASE_FROM_JAVA_VERSION == matrix.java-version && matrix.runson.name != 'macos-13' }}
```

Comment:
With this addition, `RELEASE_FROM_JAVA_VERSION` becomes less meaningful.
What about removing `RELEASE_FROM_JAVA_VERSION` and checking explicit values of matrix.runson/matrix.java-version instead?

Reply:
7b3581d0f16c2893e77d66e1c9f337d6667e0dee